src/awkward/contents/emptyarray.py

```diff
@@ -364,11 +364,12 @@ def _to_arrow(
         if options["emptyarray_to"] is None:
             return pyarrow.Array.from_buffers(
                 ak._connect.pyarrow.to_awkwardarrow_type(
-                    pyarrow.null(),
-                    options["extensionarray"],
-                    options["record_is_scalar"],
-                    mask_node,
-                    self,
+                    storage_type=pyarrow.null(),
+                    use_extensionarray=options["extensionarray"],
+                    record_is_scalar=options["record_is_scalar"],
+                    mask=mask_node,
+                    node=self,
+                    is_nonnullable_nulltype=mask_node is None,
```

Comment:
This was the key point. It's fixed now, so I'll resolve the conversation about it.